src/ImageSharp/Formats/Gif/GifEncoderCore.cs

```diff
@@ -302,16 +312,10 @@ private void WriteGraphicalControlExtension<TPixel>(ImageFrame<TPixel> imageFram
         private void WriteGraphicalControlExtension<TPixel>(ImageBase<TPixel> image, IMetaData metaData, EndianBinaryWriter writer, int transparencyIndex)
             where TPixel : struct, IPixel<TPixel>
         {
-            // TODO: Check transparency logic.
```

Comment:
This logic seems to be gone because the `DisposalMethod` has been moved to `IMetaData`. Maybe you should make the DisposalMethod nullable so you can check if you can copy the value from the original. But I do wonder if this logic makes any sense though. We can probably just have `DisposalMethod.Unspecified` as the default.